tests/util/test_networks.py

```diff
@@ -12,7 +12,79 @@
 assert_equal = functools.partial(th.testing.assert_close, rtol=0, atol=0)
 
 
-NORMALIZATION_LAYERS = [networks.RunningNorm, networks.EMANorm]
+def EMANormDecayWithinBatch(*args, **kwargs):
+    return networks.EMANorm(*args, **kwargs, decay_within_batch=True)
+
+
+NORMALIZATION_LAYERS = [
+    networks.RunningNorm,
+    networks.EMANorm,
+    EMANormDecayWithinBatch,
+]
+
+
+class EMANormIncremental(networks.EMANorm):
+    """EMA Norm with every new instance added incrementally in a for loop."""
+
+    def __init__(
+        self,
+        num_features: int,
+        decay: float = 0.99,
+        decay_within_batch: bool = False,
+        eps: float = 1e-5,
+    ):
+        """Builds EMARunningNormIncremental."""
+        super().__init__(
+            num_features,
+            decay=decay,
+            decay_within_batch=decay_within_batch,
+            eps=eps,
+        )
+
+    def update_stats(self, batch: th.Tensor) -> None:
+        """Update `self.running_mean` and `self.running_var` incrementally.
+
+        Reference Finch (2009), "Incremental calculation of weighted mean and variance".
+            (https://fanf2.user.srcf.net/hermes/doc/antiforgery/stats.pdf)
+
+        Args:
+            batch: A batch of data to use to update the running mean and variance.
+        """
+        b_size = batch.shape[0]
+        alpha = 1 - self.decay
+        if self.decay_within_batch:
+            start_index = 0
+            if self.count == 0:
+                self.running_mean = batch[0]
+                self.running_var = th.zeros_like(self.running_mean, dtype=th.float)
+                start_index = 1
+
+            for i in range(start_index, b_size):
+                diff = batch[i, ...] - self.running_mean
+                incr = alpha * diff
+                self.running_mean += incr
+                self.running_var = self.decay * (self.running_var + diff * incr)
+        else:
```

Comment:
I think this would be slightly more intuitive and easier to read using algorithm 3 rather than 2 from the pdf. That's because it basically boils down to using the incremental EMA and EMV update rules on the batch means but adding a correction factor for the batch variance.